tools/schemacode/src/bidsschematools/render/tsv.py

```diff
@@ -28,31 +30,30 @@ def fence(
     classes[:0] = ["tsv-table", "index" if linenums else "noindex"]
 
     try:
-        df = pd.read_csv(
-            io.StringIO(source),
-            sep="\t",
-            dtype=str,
-            index_col=False,
-            keep_default_na=False,
-            header=None if "noheader" in attrs else "infer",
-        )
-        md_table = tabulate(
-            df,  # type: ignore
-            tablefmt="github",
-            showindex=linenums,
-            headers="keys",
-            numalign="right",
-        )
-        html = MarkdownIt("commonmark").enable("table").render(md_table)
-        if "noheader" in attrs:
-            html = re.sub("<thead>.+</thead>", "", html, flags=re.DOTALL)
-
-        html = html.replace("<table>", f'<table class="{" ".join(classes)}">')
-
-        # Remove newlines from HTML to prevent copy-paste from inserting spaces
-        return html.replace("\n", "")
-    except Exception:
-        import traceback
-
-        exc = traceback.format_exc()
-        return f"<pre>{exc}</pre>"
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
```

Comment:
sorry -- wouldn't that ignore warnings entirely and we wouldn't know when something is off or about to be removed?

Reply:
It promotes all warnings to errors in this block.